lfshook.go

```diff
@@ -16,42 +18,107 @@ var txtFormatter = &logrus.TextFormatter{DisableColors: true}
 // Multiple levels may share a file, but multiple files may not be used for one level
 type PathMap map[logrus.Level]string
 
+// Alternatively map a log level to an io.Writer
+type WriterMap map[logrus.Level]io.Writer
+
 // Hook to handle writing to local log files.
 type lfsHook struct {
-	paths  PathMap
-	levels []logrus.Level
-	lock   *sync.Mutex
+	paths     PathMap
+	writer    WriterMap
+	levels    []logrus.Level
+	lock      *sync.Mutex
 	formatter logrus.Formatter
 }
 
 // Given a map with keys equal to log levels.
 // We can generate our levels handled on the fly, and write to a specific file for each level.
 // We can also write to the same file for all levels. They just need to be specified.
-func NewHook(levelMap PathMap) *lfsHook {
+func NewHook(levelMap interface{}) *lfsHook {
 	hook := &lfsHook{
-		paths: levelMap,
-		lock:  new(sync.Mutex),
+		lock:      new(sync.Mutex),
 		formatter: txtFormatter,
 	}
-	for level, _ := range levelMap {
-		hook.levels = append(hook.levels, level)
+
+	switch levelMap.(type) {
+	case PathMap:
+		hook.paths = levelMap.(PathMap)
+		for level := range levelMap.(PathMap) {
+			hook.levels = append(hook.levels, level)
+		}
+		break
+	case WriterMap:
+		hook.writer = levelMap.(WriterMap)
+		for level := range levelMap.(WriterMap) {
+			hook.levels = append(hook.levels, level)
+		}
+		break
```

Comment:
Should probably add a default case that panics, otherwise it is possible to get through here setting neither a pathMap nor a writerMap, and that guarantees an error case later on.